### PR TITLE
Extracting Kapt/Java Workaround from main class

### PR DESCRIPTION
**Pull request by cdsap**
Extracting workaround logic for Kapt and Java in the `RoomWorkaround`

### CODE REVIEW

**Comment by runningcode** [comment]
Does this PR change any logic or is it simply extracting classes? I wasn't sure from the description.

**Review by runningcode — COMMENTED** [review]
sorry I didn't have a chance to go through this entire PR yet

File: src/main/groovy/org/gradle/android/workarounds/room/AnnotationProcessorWorkaround.groovy [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package org.gradle.android.workarounds.room
+
+import org.gradle.android.workarounds.room.androidvariants.ApplyAndroidVariants
+import org.gradle.android.workarounds.room.task.RoomSchemaLocationMergeTask
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.tasks.TaskProvider
+
+abstract class AnnotationProcessorWorkaround<T> {
```
[the remark this answers is not in the document]
Comment (cdsap) [reply src/main/groovy/org/gradle/android/workarounds/room/AnnotationProcessorWorkaround.groovy:9]: @runningcode I run some issues with some dynamic logic in the workarounds. Will try again now and update you about the results

[the remark this answers is not in the document]
Comment (cdsap) [reply src/main/groovy/org/gradle/android/workarounds/room/AnnotationProcessorWorkaround.groovy:11]: Added `CompileStatic` for `AnnotationProcessorWorkaround`

**Review by runningcode — APPROVED** [review]
LGTM! let's merge :)

**Pull request merged**